recipes/swig/all/conanfile.py

```diff
@@ -1,170 +1,164 @@
-            autotools.flags.append("-FS")
-            # MSVC canonical names aren't understood
-            host, build = False, False
+            self.tool_requires("bison/3.8.2")
+        self.tool_requires("automake/1.16.5")
 
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            # FIXME: Apple ARM should be handled by build helpers
-            autotools.flags.append("-arch arm64")
-            autotools.link_flags.append("-arch arm64")
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-        autotools.libs = []
-        autotools.library_paths = []
+    def generate(self):
+        build_env = VirtualBuildEnv(self)
+        build_env.generate()
 
-        if self.settings.os == "Windows" and self.settings.compiler != "Visual Studio":
-            autotools.libs.extend(["mingwex", "ssp"])
+        tc = AutotoolsToolchain(self)
+        env = tc.environment()
 
-        autotools.configure(args=args, configure_dir=self._source_subfolder,
-                            host=host, build=build)
-        return autotools
+        pcre = "pcre2" if self._use_pcre2 else "pcre"
+        tc.configure_args += [
+            f"--host={self.settings.arch}",
+            "--with-swiglibdir=${prefix}/bin/swiglib",
+            f"--with-{pcre}-prefix={self.dependencies[pcre].package_folder}",
+        ]
+        if self._use_pcre2:
+            env.define("PCRE2_LIBS", " ".join("-l" + lib for lib in self.dependencies["pcre2"].cpp_info.libs))
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            tc.configure_args.append("LIBS=-ldl")
+            tc.extra_defines.append("HAVE_UNISTD_H=1")
+        elif self.settings.os == "Windows":
+            if is_msvc(self):
+                env.define("CC", "cccl -FS")
+                env.define("CXX", "cccl -FS")
+                tc.configure_args.append("--disable-ccache")
+            else:
+                tc.extra_ldflags.append("-static")
+                tc.configure_args.append("LIBS=-lmingwex -lssp")
+        elif is_apple_os(self):
+            if self.settings.arch == "armv8":
+                # FIXME: Apple ARM should be handled by build helpers
+                tc.extra_cxxflags.append("-arch arm64")
+                tc.extra_ldflags.append("-arch arm64")
+        tc.generate(env)
+
+        deps = AutotoolsDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        # Rely on AutotoolsDeps instead of pcre2-config
+        # https://github.com/swig/swig/blob/v4.1.1/configure.ac#L70-L92
+        # https://github.com/swig/swig/blob/v4.0.2/configure.ac#L65-L86
+        replace_in_file(self, os.path.join(self.source_folder, "configure.ac"),
+                        'AS_IF([test "x$with_pcre" != xno],', 'AS_IF([false],')
```

Comment:
```suggestion
        replace_in_file(self, os.path.join(self.source_folder, "configure.ac"),
                        'AS_IF([test "x$with_pcre" != xno],', 'AC_DEFINE([HAVE_PCRE], [1])\nAS_IF([false],')
```

Reply:
Hum that works on mac, but not on ubuntu/windows.

```
/usr/bin/ld: Swig/misc.o: in function `Swig_string_regex':
  misc.c:(.text+0x263b): undefined reference to `pcre2_compile_8'
  /usr/bin/ld: misc.c:(.text+0x2651): undefined reference to `pcre2_match_data_create_from_pattern_8'
```

Reply:
Oh cool, it's also affected by:
* https://github.com/conan-io/conan/issues/12784

```
cl : Command line warning D9002 : ignoring unknown option '-LIBPATH:D:/a/runner/runner/conan-cache/.conan2/p/pcre2d85e041fe43c5/p/lib'
cl : Command line warning D9002 : ignoring unknown option '-LIBPATH:D:/a/runner/runner/conan-cache/.conan2/p/bzip2e06444d88ab4f/p/lib'
cl : Command line warning D9002 : ignoring unknown option '-LIBPATH:D:/a/runner/runner/conan-cache/.conan2/p/zlib9d72e7d21adc5/p/lib'
```